packages/mui-base/src/Switch/Switch.tsx

```diff
@@ -103,6 +103,9 @@ const Switch = React.forwardRef(function Switch<RootComponentType extends React.
     elementType: Input,
     getSlotProps: getInputProps,
     externalSlotProps: slotProps.input,
+    additionalProps: {
+      role: 'switch', // Switch component should have the role "switch" to meet a11y guidelines
```

Comment:
This should be applied directly in the JSX component, no?

Reply:
I tried adding the prop into `useSwitch.getInputProps` but `useSwitch` is being used by `Radio`, `CHeckbox` along with `Switch` component. So, added it in `additionalProps`

Reply:
This should be returned by the hook.

Reply:
`useSwitch` hook is being consumed by `Radio` and `Checkbox` components, along with the `Switch` component. Adding `role = switch` to the hook returned value breaks those components. To make the hook work, we might have send another parameter to `useSwitch`, which I felt isnt an ideal solution. We have `additionalProps`, which came handy for the fix.

But, I'm open for suggestions.

Reply:
> useSwitch hook is being consumed by Radio and Checkbox components,

This is a bit unfortunate (and temporary). I'd make the hook return the correct set of props for the Switch and override them if needed in Checkbox and Radio button.
We are going to add Checkbox and Radio Button to Base UI in the next batch of components (likely late Q2 or Q3 this year), so it won't be an issue anymore.

Reply:
Got it, thanks for the update @michaldudak! Made changes as per the discussion